text/0003-measure-metric-type.md

```diff
@@ -2,50 +2,168 @@
+
+Only the gauge kind of metric supports observing the metric via a gauge `Observer` callback (as an option).  Semantically, there is an important difference between explicitly setting a gauge and observing it through a callback.  In case of setting the gauge explicitly, the call happens inside of an implicit or explicit context.  The implementation is free to associate the explicit `Set()` event with a context, for example.  When observing gauge metrics via a callback, there is no context associated with the event.
+
+As a special case, to support existing metrics infrastructure, a gauge metric may be declared as a precomputed cumulative sum using the `NonDescending` option, in which case it is defined as a strictly ascending.  The API will reject descending updates to non-descending gauges, instead submitting an SDK error event.
+
+For gauge metrics, the default OpenTelemetry implementation exports the last value that was explicitly `Set()`, or if using a callback, the current value from the Observer.
+
+### Measure metric
+
+Measure metrics express a distribution of values.  This kind of metric should be used when the count or rate of events is meaningful and either:
+
+1. The sum is of interest in addition to the count (rate)
+1. Quantile information is of interest.
+
+The key property of a measure metric event is that computing quantiles and/or summarizing a distribution (e.g., via a histogram) may be expensive.  Not only will implementations have various capabilities and algorithms for this task, users may wish to control the quality and cost of aggregating measure metrics.
+
+Like cumulative metrics, non-negative measures are an important case because they support rate calculations. As an option, measure metrics may be declared as `NonNegative`.  The API will reject negative metric events for non-negative measures, instead submitting an SDK error event.
+
+Because measure metrics have such wide application, implementations are likely to provide configurable behavior.  OpenTelemetry may provide such a facility in its standard SDK, but in case no configuration is provided by the application, a low-cost policy is specified as the default behavior, whic is to export the sum, the count (rate), the minimum value, and the maximum value.
+
+### Disable selected metrics by default
+
+All OpenTelemetry metrics may be disabled by default, as an option.  Use this option to indicate that the default implementation should be to do nothing for events about this metric.
+
+### RecordBatch API
+
+Applications sometimes want to act upon multiple metric handles in a single API call, either because the values are inter-related to each other, or because it lowers overhead.  We agree that recording batch measurements will be restricted to measure metrics, although this support could be extended to all kinds of metric in the future.
+
+A single measurement is defined as:
```

Comment:
Not sure I understand how this will be implemented:
Does the library developer need to provide Handles OR Metrics + a list of label values that will be applied to get the Handles (if an implementation decide to do that). Some other implementations may decide to record this event as it is so GetHandle can be avoided.

The second option is what we have in OpenCensus and works well.

Reply:
The problem is that I can see both cases being useful. I recognize that "raw statistics" in OC had a 3-tuple of (metric, value, pre-defined labels) as the measurement. I wanted to be able to use a 2-tuple for `RecordBatch` because I thought the purpose was to provide atomicity. 

We discussed whether it was possible to have per-call-site label values in the metrics API, and you requested treating those as labels derived from the context. This was agreed--and now a related point of confusion has come up about additional label values in the handle, which I favor.

This topic appears to be whether we can avoid the cost of allocating handles through the SDK for cases that are batch measurements that will only be used once. I see the benefit, but it seems to prevent recording a batch of metrics using a handle.

I could see supporting both options.  That's somewhat unsatisfying.

For the gRPC configuration prescribed in OC, I take it that `Handles` are used for the cumulative and gauge metrics, with pre-defined labels, but the raw measurements did not use pre-defined labels. But gRPC _knows_ the pre-defined labels here. I presume this use of pre-defined labels for cumulative and gauges is because these are pre-aggregated. Therefore, measure metrics do not benefit, the argument goes, from pre-defined labels as much.

Can we avoid this by:
(1) requiring labels supplied in a `RecordBatch` to be passed via context
(2) allowing `RecordBatch` to take either a metric instrument or a handle
(3) if a handle, the additional labels are passed by context; if a metric, both required and additional labels are passed by context.

At some point, thus blurs the distinction between call-site and per-handle labels...